agent/app/agent.go

```diff
@@ -97,6 +98,12 @@ const (
 	inServiceState                 = "InService"
 	asgLifecyclePollWait           = time.Minute
 	asgLifecyclePollMax            = 120 // given each poll cycle waits for about a minute, this gives 2-3 hours before timing out
+
+	// the size of the buffer for the metric/task health/instance health telemetry channels.
+	// By default, TCS (or TACS) will reject metrics that are older than 5 minutes (https://tiny.amazon.com/p6t8axgr).
+	// Since our metrics collection interval is currently set to 20 seconds, setting a buffer size of 15 allows us to
+	// store exactly 5 minutes of metrics in these buffers in the case where we temporarily lose connect to TCS.
```

Comment:
Is this comment fargate-specific?  I believe our ECS EC2 metrics share this channel across all running tasks on the host, so if we have 15 tasks, that will probably fill the buffer faster than 5 minutes.

Reply:
Yeah good point! I should have removed it - I was thinking of tuning this value after the channels are established.